src/common/src/config.rs

```diff
@@ -1060,6 +1059,13 @@ pub struct StreamingDeveloperConfig {
     /// A flag to allow disabling the auto schema change handling
     #[serde(default = "default::developer::stream_enable_auto_schema_change")]
     pub enable_auto_schema_change: bool,
+
+    #[serde(default = "default::developer::enable_shared_source")]
+    /// Enable shared source
+    /// If false, the shared source will be disabled,
+    /// even if session variable set.
+    /// If true, it's decided by session variable `streaming_use_shared_source` (default true)
+    pub enable_shared_source: bool,
```

Comment:
This doesn't follow the standard pattern of a session/global variable, particularly, the "cluster-level config" should be [SystemParams](https://github.com/risingwavelabs/risingwave/blob/e2c76182d0da075e8623479bc2dbe220ef23cf67/proto/meta.proto#L642-L667) instead of `RwConfig`. However, as the configuration is not user-facing but only for us, it's acceptable to me.

Reply:
> standard pattern of a session/global variable

FYI actually session variable can also be `ALTER SYSTEM` to be changed globally.

Just add this extra gate in case it's needed.